core/src/commonMain/kotlin/com/xebia/functional/xef/llm/openai/OpenAIClient.kt

```diff
@@ -0,0 +1,255 @@
+package com.xebia.functional.xef.llm.openai
+
+import com.aallam.openai.api.BetaOpenAI
```

Comment:
Is this still being moved to a different module in a subsequent PR?

Reply:
Would that new module be a dependency of core? if that is the case we can move the remaining 3 files for the OpenAI impl to a new module and link it.

Reply:
I'm not sure why that is necessary. _core_ should be independent of `open-ai`, right?

Since we have a domain to decouple _core_ from the `com.aallam.openai.api.BetaOpenAI` domain this implementation can live in a separate module that depends on _core_, right?

The other modules can then rely on this _only_ if they need the OpenAI dependency. `AIRuntime` also seems coupled to `xef-kotlin`, rather than _core_. In Java we need to think of a different strategy for dependency injection, perhaps Scala can re-use a similar idea but there we have `Using` in the Std.